[ACR]: bug fixes on token create and image import commands

yugangw-msft opened the pull request:
**Description<!--Mandatory-->**  
2 ACR command updates, see history notes below for details

**Testing Guide**  
Added e2e test for token creation
Conducted manual test on image bug fixes
**History Notes**  
[ACR] az acr token create: expose --days argument
[ACR] az acr import: accept --source argument values which contain login in server name through client end correction

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

yonzhan commented:
add to S169

qwordy commented on src/azure-cli/azure/cli/command_modules/acr/token.py:
```diff
@@ -21,12 +21,15 @@ def acr_token_create(cmd,
 
     if bool(repository_actions_list) == bool(scope_map_name):
         raise CLIError("usage error: --repository | --scope-map-name")
+    if no_passwords and days is not None:
+        raise CLIError("usage error: --no-passwords and --days are mutually exclusive.")
```
What if no_passwords is not None and days? Will CLI throw an error?

yugangw-msft replied:
if `no_password` is False or just missing, then it is a valid case to have `days` for user specified expiry time, or `9999-12-31` otherwise

yungezz commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -324,10 +324,13 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
 
+    for scope in ['acr token create', 'acr token credential generate']:
+        with self.argument_context(scope) as c:
+            c.argument('days', options_list=['--days'], help='Number of days for which the credentials will be valid. If not specified, the expiration will default to the max value "9999-12-31T23:59:59.999999+00:00"', type=int, required=False)
```
expiration_in_days? I think there's existing guideline of not putting unit in parameter name.

yugangw-msft replied:
You are right. The existing `az acr token credential generate` already uses the same name, let me update that one as well . Will put in deprecate. Since the whole command is still under preview, so the eventual breaking is fine.

yungezz commented on src/azure-cli/azure/cli/command_modules/acr/import.py:
```diff
@@ -50,10 +50,15 @@ def acr_import(cmd,
             registry = get_registry_from_name_or_login_server(cmd.cli_ctx, source_registry, source_registry)
             if registry:
+                # trim away redundant login server name, a common error
+                prefix = registry.login_server + '/'
```
how about add validation earlier at params?

yugangw-msft replied:
This change will be a bit involving, as to do it right, other 3 parameters should also be moved to the validator not just this one. Can I log an issue and do it later?

yugangw-msft replied:
Logged https://github.com/Azure/azure-cli/issues/13415

yugangw-msft commented:
@qwordy @yungezz, I have addressed the review feedback and also opened a tracking issue for improving the parameter validating. Could you merge the PR when ready; or let me know if there are other things I should address. Thanks.

yungezz commented:
hi @haroldrandom could you pls have a look at the CI failure? why the timeout is 10 mins?

haroldrandom commented:
> hi @haroldrandom could you pls have a look at the CI failure? why the timeout is 10 mins?

It passed after rerun. 
10 mins is enough for this job based on previous experience, it failed maye be caused by network jetlet occationally.
